src/win/fs.c

```diff
@@ -97,13 +97,14 @@
     return;                                                                 \
   }
 
-#define MILLION ((int64_t) 1000 * 1000)
-#define BILLION ((int64_t) 1000 * 1000 * 1000)
+#define NSEC_PER_TICK 100
+#define TICKS_PER_SEC ((long)1e9 / NSEC_PER_TICK)
+static const int64_t WIN_TO_UNIX_TICK_OFFSET = 11644473600 * TICKS_PER_SEC;
 
 static void uv__filetime_to_timespec(uv_timespec_t *ts, int64_t filetime) {
-  filetime -= 116444736 * BILLION;
-  ts->tv_sec = (long) (filetime / (10 * MILLION));
-  ts->tv_nsec = (long) ((filetime - ts->tv_sec * 10 * MILLION) * 100U);
+  filetime -= WIN_TO_UNIX_TICK_OFFSET;
+  ts->tv_sec = filetime / TICKS_PER_SEC;
```

Comment:
I don't really see why this implicit truncation is better than the previous explicit cast

Reply:
Well there is an assignment from an expression of `long long` type into a `long` variable, for sure, but the important detail here is if there is really truncation.
Truncation of a signed integer with with a simple cast like that is still UB, and UBSAN will catch that.

If we want defined behaviour, with either saturation or bit discarding assuming two's complement for example, we have to do something different.

Reply:
IIRC, this field `uint64_t` for clients that already upgraded to libuv v2 (though it is unreleased) to avoid this exact problem -- as indeed it will become a significant problem in about a decade for existing clients of v1

Reply:
Just finished reviewing. Can you split out this commit to a separate PR? The rest look good to land as a batch

Reply:
Done, will push a new PR after this one is merged, since there is a dependency and I can't stack them on GitHub without access to create branches on this repository.